### PR TITLE
Make Integration Tests runnable in Console Launcher (#197)

### PR DESCRIPTION
**Pull request by dimas-b**
* Assemble a test jar with dependencies to simplify
  usage with Junit Console Launcher

* Minor usability fixes in StorageContainer and ExternalResource

### CODE REVIEW

File: DEV_GUIDE.md [hunk 1]
```diff
@@ -185,11 +185,82 @@ is managed manually, use the following options to convey connection information
 * `-Dstargate.test.backend.use.external=true`
 * `-Dstargate.test.backend.cluster_name=<CLUSTER_NAME>`
 * `-Dstargate.test.backend.dc=<DATA_CENTER_NAME>`
-* `-Dstargate.test.backend.nodes=<NUMBER_OR_STORAGE_NODES>`
+
+When integration test run in with debugging options, the related Stargate nodes will also be
```
Comment (dougwettlaufer) [inline DEV_GUIDE.md:189]:
```suggestion
When integration test run with debugging options, the related Stargate nodes will also be
```

File: DEV_GUIDE.md [hunk 2]
```diff
@@ -185,11 +185,82 @@ is managed manually, use the following options to convey connection information
...
+  of the Stargate nodes.
+  When this extension is active, it will automatically inject test method parameters of type
+  `StargateConnectionInfo` and `StargateEnvironmentInfo`.
+
+Integration tests that do not need Stargate nodes (e.g. `CassandraPersistenceIT`) can use only
+the `ExternalStorage` extension by having the `@ExtendWith(ExternalStorage.class)` annotation
+either directly on the test class or on one of its super-classes.
+
+Integration tests that need both storage and Stargate nodes, should use the `@UseStargateContainer`
+annotation to active both extensions in the right order.
```
Comment (dougwettlaufer) [inline DEV_GUIDE.md:250]:
```suggestion
annotation to activate both extensions in the right order.
```

File: DEV_GUIDE.md [hunk 3]
```diff
@@ -185,11 +185,82 @@ is managed manually, use the following options to convey connection information
...
+
+The code element holding `@ClustgerSpec` or `@StargateSpec` annotations controls the lifecycle of
+the nodes they define. If the "spec" is present at the class level (inherited), the corresponding 
+nodes will be started/stopped according to `@BeforeAll` / `@AfterAll` JUnit 5 callbacks. Similarly,
+if the spec is present at the method level, the nodes' lifecycle will follow `@BeforeEach` /
+`@AfetrEach` callbacks. An exception to this rule is when the spec has the `shared` property set
```
Comment (dougwettlaufer) [inline DEV_GUIDE.md:256]:
```suggestion
`@AfterEach` callbacks. An exception to this rule is when the spec has the `shared` property set 
```

**Review by dougwettlaufer — APPROVED** [review]

**Review by mpenick — APPROVED** [review]
+1 Awesome docs.

**Comment by dimas-b** [comment]
resolves #197

**Pull request merged**